modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImplTest.java

```diff
@@ -0,0 +1,683 @@
+            ColumnDescriptor actualColumnDescriptor = descriptor.columnDescriptor(i);
+
+            checkColumns(actualColumnDescriptor, i, expectedColumnDescriptor);
+        }
+    }
+
+    private static Stream<Arguments> systemViewColumnTypes() {
+        List<Arguments> allArgs = new ArrayList<>();
+
+        for (SystemViewType type : SystemViewType.values()) {
+            columnTypes().map(args -> {
+                Object[] vals = args.get();
+
+                Object[] newVals = new Object[vals.length + 1];
+                newVals[0] = type;
+
+                System.arraycopy(vals, 0, newVals, 1, vals.length);
+
+                return Arguments.of(newVals);
+            }).forEach(allArgs::add);
+        }
+
+        return allArgs.stream();
+    }
+
+    private static Stream<Arguments> systemViewDistributions() {
+        return Stream.of(
+                Arguments.of(SystemViewType.NODE, IgniteDistributions.identity(0)),
+                Arguments.of(SystemViewType.CLUSTER, IgniteDistributions.single())
+        );
+    }
+
+
+    private static IgniteSystemView getSystemView(IgniteSchema schema, String name) {
+        Table systemViewTable = schema.getTable(name);
+        assertNotNull(systemViewTable);
+
+        IgniteSystemView systemView = assertInstanceOf(IgniteSystemView.class, systemViewTable);
+        assertEquals(systemView.name(), name);
+
+        return systemView;
+    }
+
+    private static IgniteSchema unwrapSchema(SchemaPlus schemaPlus) {
+        IgniteSchema igniteSchema = schemaPlus.unwrap(IgniteSchema.class);
+        assertNotNull(igniteSchema);
+        return igniteSchema;
+    }
+
+    private static IgniteTable getTable(IgniteSchema schema, String name) {
+        IgniteTable table = (IgniteTable) schema.getTable(name);
+        assertNotNull(table);
+        return table;
+    }
+
+    private static Stream<Arguments> columnTypes() {
+        return Stream.of(
+                Arguments.of(ColumnType.BOOLEAN, -1, -1),
+                Arguments.of(ColumnType.INT8, -1, -1),
+                Arguments.of(ColumnType.INT16, -1, -1),
+                Arguments.of(ColumnType.INT32, -1, -1),
+                Arguments.of(ColumnType.INT64, -1, -1),
+                Arguments.of(ColumnType.FLOAT, -1, -1),
+                Arguments.of(ColumnType.DOUBLE, -1, -1),
+                Arguments.of(ColumnType.DECIMAL, 4, 0),
+                Arguments.of(ColumnType.DECIMAL, 4, 2),
+                Arguments.of(ColumnType.NUMBER, 4, -1),
+                Arguments.of(ColumnType.STRING, 40, -1),
+                Arguments.of(ColumnType.BYTE_ARRAY, 40, -1),
+                Arguments.of(ColumnType.DATE, -1, -1),
+                Arguments.of(ColumnType.TIME, 2, -1),
+                Arguments.of(ColumnType.DATETIME, 2, -1),
+                Arguments.of(ColumnType.TIMESTAMP, 2, -1),
+                Arguments.of(ColumnType.UUID, 2, -1),
```

Comment:
There was a typo in original code - UUID does not have precision, so it should be `Arguments.of(ColumnType.UUID, -1, -1),`.

Reply:
done

Reply:
done